Parsers/ASimWebSession/Parsers/ASimWebSessionCiscoFirepower.yaml

```diff
@@ -0,0 +1,197 @@
+      "Archive Block-Encrypted", "Encrypt", "Failure",
+      "Archive Block-Failed to Inspect", "Deny", "Failure"
+  ];
+  let DirectionLookup = datatable (CommunicationDirection: string, NetworkDirection: string)[
+      "1", "Inbound",
+      "2", "Outbound"
+  ];
+  let parser=(disabled: bool=false) {
+      let alllogs=CommonSecurityLog
+          | where not(disabled) and DeviceVendor == "Cisco" and DeviceProduct == "Firepower"
+          | where DeviceEventClassID in("File:500:1", "FileMalware:502:1", "FireAMP:125:1")
+          | parse-kv AdditionalExtensions as (start: long) with (pair_delimiter=';', kv_delimiter='=');
+      let Fireamplogs = alllogs
+          | where DeviceEventClassID == "FireAMP:125:1"
+          | extend
+              EventMessage = DeviceCustomString5,
+              ThreatId = EventOutcome,
+              ThreatName = DeviceCustomString2
+          | extend AdditionalFields = bag_pack(
+                                  "policy",
+                                  DeviceCustomString1,
+                                  "process",
+                                  SourceProcessName,
+                                  "connectionInstance",
+                                  ProcessID,
+                                  "RequestClientApplication",
+                                  RequestClientApplication
+                              )
+          | project-rename DstUsername = DestinationUserName
+          | extend
+              DstUsernameType = _ASIM_GetUsernameType(DstUsername),
+              DstUserType = _ASIM_GetUserType(DstUsername, "");
+      let Filelogs = alllogs
+          | where DeviceEventClassID in ("File:500:1", "FileMalware:502:1")
+          | extend AdditionalFields = bag_pack(
+                                  "connectionInstance",
+                                  ProcessID,
+                                  "signaturedata",
+                                  DeviceCustomString4,
+                                  "RequestClientApplication",
+                                  RequestClientApplication
+                              );
+      union Fireamplogs, Filelogs
+      | invoke _ASIM_ResolveNetworkProtocol('Protocol')
+      | extend NetworkProtocol = iff(NetworkProtocol == "Unassigned" and Protocol !in (63, 68, 99, 114, 253, 254), Protocol, NetworkProtocol)
+      | lookup DirectionLookup on CommunicationDirection
+      | lookup EventFieldsLookup on DeviceAction
+      | extend
+          EventStartTime = coalesce(unixtime_milliseconds_todatetime(start), unixtime_milliseconds_todatetime(tolong(ReceiptTime))),
+          DstIpAddr = coalesce(DestinationIP, DeviceCustomIPv6Address3),
+          SrcIpAddr = coalesce(SourceIP, DeviceCustomIPv6Address2),
+          EventSeverity = case(
+                      DvcAction == "Allow" and DeviceCustomString3 =~ "Malware",
+                      "High",
+                      DvcAction == "Deny" and DeviceCustomString3 =~ "Malware",
```

Comment:
Can we also do
DvcAction == "Allow" and isnotempty (DeviceCustomString3), "Medium"

Reply:
As discussed,We cannot add above condition because possible values for DeviceCustomString3(disposition) are as below:
![image](https://github.com/Azure/Azure-Sentinel/assets/107753570/6c6ce065-759e-4bf7-a5dd-812fa0a7b6b5)